twisted/internet/defer.py

```diff
@@ -661,6 +661,66 @@ def __str__(self):
     __repr__ = __str__
 
 
+    def __await__(self):
+        awaiter = _MakeDeferredAwaitable(self)
+        return awaiter.__await__()
+
+
+
+class _MakeDeferredAwaitable(object):
+    """
+    A subgenerator which is created by L{Deferred.__await__}.
+    """
+    result = _NO_RESULT
+
+    def __init__(self, d):
+        self.d = d
+        self.d.addBoth(self._setResult)
+
+    def _setResult(self, value):
+        self.result = value
+
+    def __next__(self):
+        if self.result is not _NO_RESULT:
+            raise StopIteration(self.result)
+        return self.d
+
+    def __await__(self):
```

Comment:
Right. Should this actually need to implement `__await__` at all? Since Deferred is the thing with await, I just need to change it to call `__iter__` and rename the `__await__` on this object...